docs/uenv-compilation-spack.md

```diff
@@ -0,0 +1,163 @@
+
+The uenv of a supported application contains all the dependencies to build the software with a particular configuration. In [Spack], such configuration is defined by a [spec] (see [Spack Basic Usage] for more details). Most uenvs already provide modules or [Spack Filesystem Views] which allow to manually build the same configuration. However, it is possible that you want to build or develop an application with a different configuration. To avoid re-building the whole uenv, you can re-use what is already there and build your new configuration using [Spack]. 
+
+This guide explains a _developer workflow_ allowing to either build your own package with [Spack], or use [Spack] to build all the dependencies and provide a build environment for the package.
+
+!!! note
+    This guide assumes that you have a local installation of [Spack]. If you don't have [Spack] installed, follow [Spack Getting Started].
+
+!!! warning
+    Avoid installing [Spack] on `HOME`. Packages are installaed within the `spack/` folder, and you might quickly run out of space.
+
+## Example: CP2K
+
+As an example, we will consider [CP2K] as the application we want to develop. While the uenv for [CP2K] we provide is rather complete, let's assume for simplicity that the provided configuration is very bare bone:
+
+```
+cp2k@2024.1 +cuda cuda_arch=80
+```
+
+This [spec] defines a build of [CP2K] version `2024.1` with CUDA acceleration. All other dependencies and features are defined by the default values in the [CP2K Spack package].
+
+## Set uenv as upstream Spack instance
+
+Activate the uenv. Here we assume the uenv described above is called `cp2k/2024.1` and it is already deployed:
+
+```bash
+uenv image pull cp2k/2024.1
+uenv start cp2k/2024.1
+```
+
+With the uenv active, we can now tell our local [Spack] instance to use the uenv as an upstream [Spack] instance (see [Chaining Spack Installations] for more details):
+
+```bash
+export SPACK_SYSTEM_CONFIG_PATH=/user-environment/config/
+```
+
+!!! note
+    We assumed here that the uenv is mounted in the standard location `/user-environment/`. If it is mounted in a non-standard location, adjust the previous command accordingly.
+
+## Building your own version 
+
+Let's assume we want to have a version of [CP2K] what uses the COSMA library for communication-optimal matrix multiplication with GPU-aware MPI:
+
+```
+cp2k@2024.1 +cuda cuda_arch=80 +cosma ^cosma +gpu_direct
+```
+
+COSMA is not availabe in the uenv described above since it is not a dependency needed by `cp2k@2024.1 +cuda cuda_arch=80`. 
+
+### Spack Environment
+
+To make things clean and reproducible, we use [Spack Environments] do describe what we want to build. To define a [Spack environment] we create the following file in a folder (hereafter referred to as `SPACK_ENV_FOLDER`):
```

Comment:
It is missing that the file must be named `spack.yaml`.

What about giving the commands `spack env create`, `spack  -e [..] add <spec>`
`spack config add [..]` to create `spack.yaml`?

Reply:
In addition to `spack env create <name>`, you can also do `spack -e <name> config edit` to dump the below contents into the file. Not sure which is simpler to grok for someone not used to envs though.